internal/controller/processor_test.go

```diff
@@ -575,574 +817,1025 @@ var _ = Describe("AppConfiguationProvider processor", func() {
+			cachedNextFeatureFlagRefreshReconcileTime := processor.ReconciliationState.NextFeatureFlagRefreshReconcileTime
+
+			_ = processor.PopulateSettings(&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: fakeResourceVersion,
+			}}, existingSecrets)
+
+			_, _ = processor.Finish()
+
+			Expect(processor.ReconciliationState.ExistingSecretReferences).Should(Equal(allSettingsReturnedByKeyValueRefresh.SecretReferences))
+			Expect(processor.ReconciliationState.KeyValueETags[acpv1.Selector{
+				KeyFilter: &testKey,
+			}]).Should(Equal(updatedKeyValueEtags[acpv1.Selector{
+				KeyFilter: &testKey,
+			}]))
+			Expect(processor.ReconciliationState.FeatureFlagETags[acpv1.Selector{
+				KeyFilter: &wildcard,
+			}]).Should(Equal(updatedFeatureFlagEtags[acpv1.Selector{
+				KeyFilter: &wildcard,
+			}]))
+			Expect(processor.ReconciliationState.NextKeyValueRefreshReconcileTime).Should(Equal(expectedNextKeyValueRefreshReconcileTime))
+			Expect(processor.ReconciliationState.NextFeatureFlagRefreshReconcileTime).Should(Equal(cachedNextFeatureFlagRefreshReconcileTime))
+
+			processor.CurrentTime = metav1.NewTime(processor.ReconciliationState.NextSecretReferenceRefreshReconcileTime.Time.Add(5 * time.Second))
+			processor.RefreshOptions = &RefreshOptions{}
+
+			allSettingsReturnedBySecretRefresh := &loader.TargetKeyValueSettings{
+				SecretSettings: map[string]corev1.Secret{
+					secretName: {
+						Data: secretResult,
+						Type: corev1.SecretType("Opaque"),
+					},
+				},
+				SecretReferences: map[string]*loader.TargetSecretReference{
+					secretName: {
+						Type:            corev1.SecretType("Opaque"),
+						SecretsMetadata: secretMetadata,
+					},
+				},
+			}
+
+			// only secret refresh
+			mockConfigurationSettings.EXPECT().CheckPageETags(gomock.Any(), gomock.Any()).Return(false, nil).AnyTimes()
+			mockConfigurationSettings.EXPECT().ResolveSecretReferences(gomock.Any(), gomock.Any(), gomock.Any()).Return(allSettingsReturnedBySecretRefresh, nil)
+			expectedNextSecretReferenceRefreshReconcileTime := metav1.NewTime(processor.CurrentTime.Time.Add(1 * time.Minute))
+
+			_ = processor.PopulateSettings(&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: fakeResourceVersion,
+			}}, existingSecrets)
+
+			_, _ = processor.Finish()
+
+			Expect(processor.ReconciliationState.ExistingSecretReferences).Should(Equal(allSettingsReturnedBySecretRefresh.SecretReferences))
+			Expect(processor.ReconciliationState.KeyValueETags[acpv1.Selector{
+				KeyFilter: &testKey,
+			}]).Should(Equal(updatedKeyValueEtags[acpv1.Selector{
+				KeyFilter: &testKey,
+			}]))
+			Expect(processor.ReconciliationState.FeatureFlagETags[acpv1.Selector{
+				KeyFilter: &wildcard,
+			}]).Should(Equal(updatedFeatureFlagEtags[acpv1.Selector{
+				KeyFilter: &wildcard,
+			}]))
+			Expect(processor.ReconciliationState.NextSecretReferenceRefreshReconcileTime).Should(Equal(expectedNextSecretReferenceRefreshReconcileTime))
```

Comment:
Assert the `NextKeyValueRefreshReconcileTime` and `NextFeatureFlagRefreshReconcileTime` not been updated, still keep the previous `expectedNextKeyValueRefreshReconcileTime` and `cachedNextFeatureFlagRefreshReconcileTime` value.